File: apps/spring-boot-api/src/main/java/org/bootstrapbugz/api/shared/error/exception/ResourceNotFoundException.java
package org.bootstrapbugz.api.shared.error.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.io.Serial;

@Getter
public class ResourceNotFoundException extends RuntimeException {
  @Serial private static final long serialVersionUID = -6147521296995365840L;
  private final HttpStatus status;

  public ResourceNotFoundException(String message) {
    super(message);
    this.status = HttpStatus.NOT_FOUND;
  }
}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/shared/error/exception/ResourceNotFoundException.kt
package org.bootstrapbugz.api.shared.error.exception

import org.springframework.http.HttpStatus
import java.io.Serial

class ResourceNotFoundException(message: String?) : RuntimeException(message) {
    val status: HttpStatus = HttpStatus.NOT_FOUND

    companion object {
        @Serial
        private val serialVersionUID = -6147521296995365840L
    }
}
